core/src/main/java/org/eclipse/krazo/core/HttpCommunicationUnwrapper.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2014-2015 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2018-2020 Eclipse Krazo committers and contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.eclipse.krazo.core;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Defines an SPI which allows the client to unwrap a HTTP request or response
+ * into a standard {@link javax.servlet.http.HttpServletRequest} or {@link javax.servlet.http.HttpServletResponse}.
+ * <p>
+ * This SPI is intended to be used in case a JAX-RS implementation uses its own wrapper classes for HTTP requests and responses.
+ */
+public interface HttpCommunicationUnwrapper {
```

Comment:
Not 100% sure I like the name. But naming is hard. :-)

Maybe something like `ServletUnwrapper`, `ServletClassUnwrapper`? Not sure if this is better though. :-)

Reply:
Hmmm I'm struggling with this naming too, but I think `Servlet...` indicates, that we work with servlets and not HTTP requests and responses :thinking:

Reply:
I agree. While the MVC spec doesn't directly depend on Servlet, Krazo only works in Servlet environments at the moment. Therefore, using Servlet in the name may be ok. However, I'm not strong on this one. So let's keep it as it is.